def.bzl

```diff
@@ -176,21 +205,52 @@ def gazelle(name, **kwargs):
         kwargs.pop("args")
 
     visibility = kwargs.pop("visibility", default = None)
+    return name + "-runner", visibility
+
+def gazelle(name, **kwargs):
+    runner_name, visibility = _gazelle_kwargs_prepare(name, kwargs)
 
+    _gazelle_runner(
+        name = runner_name,
+        **kwargs
+    )
     tags_set = {t: "" for t in kwargs.pop("tags", [])}
     tags_set["manual"] = ""
     tags = [k for k in tags_set.keys()]
-    runner_name = name + "-runner"
-    _gazelle_runner(
-        name = runner_name,
+    native.sh_binary(
+        name = name,
         tags = tags,
+        srcs = [runner_name],
+        visibility = visibility,
+        deps = ["@bazel_tools//tools/bash/runfiles"],
+        data = kwargs["data"] if "data" in kwargs else [],
+    )
+
+def gazelle_test(name, no_sandbox = False, **kwargs):
+    if not no_sandbox:
```

Comment:
What are we using this argument for? It's seemed to be forced to `True`.

Reply:
`no-sandbox` is a nonstandard operation. This does provide clear semantics.

Reply:
I get your point. I'm still worried about whether this could end up creating more confusion than it prevents since the attribute seems very similar to the `no-sandbox` tag, but is something else entirely. 

If we think that the rule is sufficiently hermetic to be useful and also make it somewhat clear that it doesn't specify it's dependencies (via documentation and also just by the simple fact that users aren't specifying any on the macro), maybe we just don't need it?

Reply:
You convinced me. My initial idea was to align with the `buildifier_test` rule of buildtools and the `format_test` of rules_lint, but unlike these two rules, `gazelle_test` did not provide a non-workspace mode. So no_sandbox is a bit redundant.